src/image/digest.rs

```diff
@@ -0,0 +1,167 @@
+//! Functionality corresponding to <https://github.com/opencontainers/image-spec/blob/main/descriptor.md#digests>.
+
+/// The well-known identifier for a SHA-256 digest.
+/// At this point, no one is really using alternative digests like sha512, so we
+/// don't yet try to expose them here in a higher level way.
+const ALG_SHA256: &str = "sha256";
+
+fn char_is_lowercase_ascii_hex(c: char) -> bool {
+    matches!(c, '0'..='9' | 'a'..='f')
+}
+
+/// algorithm-component ::= [a-z0-9]+
+fn char_is_algorithm_component(c: char) -> bool {
+    matches!(c, 'a'..='z' | 'A'..='Z' | '0'..='9')
+}
+
+/// encoded ::= [a-zA-Z0-9=_-]+
+fn char_is_encoded(c: char) -> bool {
+    char_is_algorithm_component(c) || matches!(c, '=' | '_' | '-')
+}
+
+/// A parsed pair of algorithm:digest as defined
+/// by <https://github.com/opencontainers/image-spec/blob/main/descriptor.md#digests>
+pub struct Digest<'a> {
+    /// The algorithm name (e.g. sha256, sha512)
+    pub algorithm: &'a str,
+    /// The algorithm component (lowercase hexadecimal)
+    pub value: &'a str,
+}
```

Comment:
Do we want to use `derive` in the same way as for other structures within this crate? Like the getters?

Reply:
I think it makes total sense to only expose a getter to disallow (accidental) mutation of the values to become invalid states (though, IMO, you'd rather have to *work* to make that accident happen; I don't know what a programmer would be thinking to end up with `&mut Digest`).

I battled the `getset` crate for a bit and I couldn't figure out how to convince it not to add an extra `&` reference on the getter, so I ended up just doing getters manually. (And dropped the `pub` from the values).